pkg/controller/kogitoapp/integration/integration.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/util"
+	v1 "github.com/openshift/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"reflect"
+)
+
+const (
+	kogitoDataIndexRouteEnv = "KOGITO_DATAINDEX_URL"
+)
+
+// InjectEnvVarsFromExternalServices inject environment variables from external services that the KogitoApp runtime might need
+func InjectEnvVarsFromExternalServices(kogitoApp *v1alpha1.KogitoApp, dc *v1.DeploymentConfig, client *client.Client) error {
+	// We look for a deployed data index to inject into the runtime service
+	// later we could also integrate with other external services like Kafka, Infinispan and SSO
+	route, err := getKogitoDataIndexRoute(client, kogitoApp.GetNamespace())
+	if err != nil {
+		return err
+	}
+
+	if len(dc.Spec.Template.Spec.Containers) > 0 {
+		oldRoute := util.GetEnvVarFromContainer(kogitoDataIndexRouteEnv, dc.Spec.Template.Spec.Containers[0])
```

Comment:
I don't think it is necessary to compare with the old value here, only need to describe the expected deployment. The comparison with the deployed DC will be done in [resource_comparator.go](https://github.com/kiegroup/kogito-cloud-operator/blob/0569bc0a936135dcc44e5340d5c0564aae84c393/pkg/controller/kogitoapp/resource/resource_comparator.go#L57) . 

Everything else looks good to me. Thanks!

Reply:
Yeah, I was in doubt wether or not adding the comparison in here to not update the resource of not needed. I’ll run some tests removing this line: deleting the data index should remove the variable.